2017/ru/0xa5-broken-access-control.md

```diff
@@ -0,0 +1,71 @@
+
+## Является ли приложение уязвимым?
+
+Контроль доступа предполагает наличие политики, определяющей права пользователей. Обход ограничений доступа обычно приводит к несанкционированному разглашению, изменению или уничтожению данных, а также выполнению непредусмотренных полномочиями бизнес-функций. Наиболее распространенные уязвимости контроля доступа включают:
+
+* обход ограничений доступа путем изменения URL, внутреннего состояния приложения или HTML-страницы, а также с помощью специально разработанных API;
+* возможность изменения первичного ключа для доступа к записям других пользователей, включая просмотр или редактирование чужой учетной записи;
+* повышение привилегий. Выполнение операций с правами пользователя, не входя в систему, или с правами администратора, войдя в систему с правами пользователя;
+* манипуляции с метаданными, например, повторное воспроизведение или подмена токенов контроля доступа JWT или куки-файлов, а также изменение скрытых полей для повышения привилегий или некорректное аннулирование JWT;
+* несанкционированный доступ к API из-за некорректной настройки междоменного использования ресурсов;
+* доступ неаутентифицированных пользователей к страницам, требующим аутентификации, или доступ непривилегированных пользователей к привилегированным страницам. Доступ к API с отсутствующим контролем привилегий для POST, PUT и DELETE.
+
+## Как предотвратить
+
+Контроль доступа эффективен только при реализации через проверенный код на стороне сервера или беcсерверный API, где атакующий не может изменять проверки прав доступа или метаданные. Рекомендуется:
+
+* запрещать доступ по умолчанию, за исключением открытых ресурсов;
+* реализовать механизмы контроля доступа и использовать их во всех приложениях, а также минимизировать междоменное использование ресурсов;
+* контролировать доступ к моделям, используя владение записями, а не возможность пользователей создавать, просматривать, обновлять или удалять любые записи;
+* использовать модели доменов для реализации специальных ограничений, относящихся к приложениям;
+* отключить вывод списка каталогов веб-сервера, а также обеспечить отсутствие метаданных файлов (например, .git) и файлов резервных копий в корневых веб-каталогах;
+* регистрировать сбои контроля доступа и уведомлять администраторов при необходимости (например, если сбои повторяются);
+* ограничивать частоту доступа к API и контроллерам для минимизации ущерба от инструментов автоматизации атак;
+* аннулировать токены JWT на сервере после выхода из системы.
+
+Разработчики и контролеры качества должны проводить функциональную проверку контроля доступа и тестировать интеграцию.
+
+## Примеры сценариев атак
+
+**Сценарий №1**: Приложение использует непроверенные данные в SQL-вызове, который обращается к информации об учетной записи:
+
+```
+  pstmt.setString(1, request.getParameter("acct"));
```

Comment:
You use indent here and below. Why? It's redundant.

Reply:
Just copied from [original files](https://github.com/OWASP/Top10/tree/master/2017/en).